### DISCUSSION TITLE
Doe this work on a Mac ?

### DISCUSSION THREAD

**Discussion by thecontinium** [NeogitOrg/neogit#776] [General]
I raised this [issue](https://github.com/NeogitOrg/neogit/issues/476) back in March and tried again today.
With all the updates ( in plugins and neovim since March  ) I am still getting the same error.

Is this a Mac OS specific issue or is it something to do with my environment ?

Cheers

**Comment by CKolkey** [comment]
Hm. I've been using this on a Mac daily for the last year without issue, so I suspect it's to do with your environment.

**Reply by thecontinium** [reply]
Thanks - I thought it might be.
Strangely it happens on 3 Mac's I have !
Now to figure out why - any thoughts ?

**Reply by CKolkey** [reply]
No, but if you have some dotfiles i could peek at I can try :)

**Reply by thecontinium** [reply]
Thanks for the offer - I'll root around first and see it I can get somewhere. I may be back :)

**Reply by thecontinium** [reply]
In my .zshenv this causes the problem:

``` sh
export XDG_CACHE_HOME=$HOME/.cache
```

If I replace $HOME with /Users/... it works.

Something to do with the headless nvim using XDG_CACHE_HOME and $HOME not being defined for a non interactive session ?

**Reply by CKolkey** [reply]
Can `~` replace $HOME?

**Reply by thecontinium** [reply]
That turned out to be a red herring. 
The offending configuration was setting XDG_RUNTIME_DIR with out creating the directory.
``` sh
export XDG_RUNTIME_DIR=/tmp/run
```
This fixes the issue:
``` sh
mkdir -p $XDG_RUNTIME_DIR
```

**Reply by CKolkey** [reply]
Great! Glad it's not on our end 😅

**Reply by thecontinium** [reply]
Needed a few spare hours to find but at last I can get to use Neogit in anger. 
Good to see you guys are actively maintaining the plugin.
Thanks.
